### DISCUSSION TITLE
Derive a class from GridSplitter?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7491] [Q&A]
I'd like to derive a class directly from GridSplitter for use in AXAML in order to add a little custom functionality.

While I can do this, the splitter appear dead at runtime and does not respond.

Anyway I can do this (or alternative i.e. as a UserControl)?

**Comment by jp2masa** [comment]
You probably need to override `IStyleable.StyleKey` like this: https://github.com/AvaloniaUI/Avalonia/issues/3812#issuecomment-618365969.

**Comment by kuiperzone** [comment]
Cheers
